core/lib/types/src/storage/mod.rs

```diff
@@ -110,6 +110,30 @@ pub fn get_is_account_key(account: &Address) -> StorageKey {
     StorageKey::new(deployer, key)
 }
 
+pub fn get_immutable_simulator_key(account: &Address, immutable_key: H256) -> StorageKey {
+    // All the immutable variables are stored as inside the ImmutableSimulator system contracts
+    // in a mapping of the following form:
+    // mapping(uint256 contractAddress => mapping(uint256 index => bytes32 value)) internal immutableDataStorage;
+
+    let immutable_simulator = AccountTreeId::new(IMMUTABLE_SIMULATOR_STORAGE_ADDRESS);
+
+    let padded_address = address_to_h256(account);
+
+    let inner_mapping_slot = keccak256(
```

Comment:
Nit: There's `web3::keccak256_concat` function that implements this functionality.